src/server/game/Maps/Map.cpp

```diff
@@ -3252,6 +3252,21 @@ void Map::SendZoneDynamicInfo(Player* player)
     }
 }
 
+void Map::PlayDirectSoundToMap(uint32 soundId, uint32 zoneId)
+{
+    Map::PlayerList const& players = GetPlayers();
+    if (!players.isEmpty())
+    {
+        WorldPacket data(SMSG_PLAY_SOUND, 4);
+        data << uint32(soundId);
+
+        for (Map::PlayerList::const_iterator itr = players.begin(); itr != players.end(); ++itr)
+            if (Player* player = itr->GetSource())
+                if (!zoneId || player->GetZoneId() == zoneId)
+                    player->SendDirectMessage(&data);
```

Comment:
Can you explain what does this do? I Don't quite understand it 😕

Reply:
Basically it sends a sound to all the players in a certain zone in the map. It just needs a packet to work.